src/pages/trainer/training/edit/actions/updateEditor.ts

```diff
@@ -0,0 +1,15 @@
+import {trainerActionEnums} from '../../../../../common/actionEnums/trainer';
+import {textAreaTool} from '../../../../../common/ui/tools/textAreaTool';
+
+export const updateEditorAction = (editor: HTMLTextAreaElement, caret: string, offset: number) => {
+  const contentWithCaret = textAreaTool.insertAtCaretGetText(editor, caret, offset);
+  const cursorStartPosition = textAreaTool.caculateStartCursorPositionPlusOffset(editor, offset);
```

Comment:
Then, we need to pass some object of type (ITextAreaFields or something) to textAreaTool with necessary textarea fields to work with. 
Then refresh textArea state via reducer